sdk/storage/azure-storage-blob/v10_v12.md

```diff
@@ -0,0 +1,437 @@
+# Storage Blob Service SDK Migration Guide from 10.x/11.x to 12.x
+
+In this section, we list main changes you need to be aware of when converting your Storage Blob SDK library from Version 10/Version 11 to Version 12.
+
+## Prerequisites
+Java Development Kit (JDK) with version 8 or above
+
+## Reactor and ReactorNetty
+We've updated our asynchronous framework from RxJava to Reactor.
+For more info related to Reactor, please refer to [Project Reactor](https://projectreactor.io/)
+For more info of the motivation behind this major change, please refer to [this guide](https://github.com/Azure/azure-storage-java/blob/master/V12%20Upgrade%20Story.md).
+
+## Converting Core Classes
+Our core asynchronous classes have been replaced, as well as new synchronous counterparts added.
+| Core v10/11 classes  | Equivalent v12 Class | NEW synchronous clients |
+|---|---|---|
+| ServiceURL  |  BlobServiceAsyncClient  | BlobServiceClient  |
+| ContainerURL |  ContainerAsyncClient | ContainerClient |
+| BlobURL    | BlobAsyncClient  |   BlobClient |
+| BlockBlobURL  |  BlockBlobAsyncClient  |  BlockBlobClient  |
+| AppendBlobURL  |  AppendBlobAsyncClient | AppendBlobClient  |
+| PageBlobURL |  PageBlobAsyncClient | PageBlobClient  |
```

Comment:
I think we should mention that the TransferManager methods were put on BlobAsyncClient and BlobClient so people know where to look for file transfer operations